src/module_wrap.cc

```diff
@@ -632,10 +638,27 @@ Maybe<URL> ResolveDirectory(Environment* env,
 
 }  // anonymous namespace
 
-Maybe<URL> Resolve(Environment* env,
+Maybe<URL> ResolveRecurse(Environment* env,
```

Comment:
This was done to make a separate code path for resolving dependencies than local modules. Not 100% happy with the current implementation... but there is definitely a difference in the algorithm now (still need to resolve main for modules).

Open to suggestions on better ways to do this

Reply:
If this is renamed to `ResolveRelativeWithExtensions`, and used only by main lookups, then the `ShouldBeTreatedAsRelativeOrAbsolutePath` check can be skipped.